android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java

```diff
@@ -0,0 +1,359 @@
+                callback.reject("invalid", "Couldn't remove an invalid list of tracks");
+            } else if(indexes.isEmpty()) {
+                callback.reject("invalid", "No tracks found");
+            } else if (indexes.size() == 1) {
+                binder.getPlayback().remove(indexes.get(0), callback);
+            } else {
+                binder.getPlayback().remove(indexes, callback);
+            }
+        });
+    }
+
+    @ReactMethod
+    public void skip(final String track, final Promise callback) {
+        waitForConnection(() -> binder.getPlayback().skip(track, callback));
+    }
+
+    @ReactMethod
+    public void skipToNext(final Promise callback) {
+        waitForConnection(() -> binder.getPlayback().skipToNext(callback));
+    }
+
+    @ReactMethod
+    public void skipToPrevious(final Promise callback) {
+        waitForConnection(() -> binder.getPlayback().skipToPrevious(callback));
+    }
+
+    @ReactMethod
+    public void reset() {
+        waitForConnection(() -> binder.getPlayback().reset());
+    }
+
+    @ReactMethod
+    public void play() {
+        waitForConnection(() -> binder.getPlayback().play());
+    }
+
+    @ReactMethod
+    public void pause() {
+        waitForConnection(() -> binder.getPlayback().pause());
+    }
+
+    @ReactMethod
+    public void stop() {
+        waitForConnection(() -> binder.getPlayback().stop());
+    }
+
+    @ReactMethod
+    public void seekTo(final double seconds) {
+        waitForConnection(() -> binder.getPlayback().seekTo(Utils.toMillis(seconds)));
+    }
+
+    @ReactMethod
+    public void setVolume(final float volume) {
+        Log.e(Utils.LOG, "Updating the volume is currently unsupported");
+    }
+
+    @ReactMethod
+    public void getVolume(final Promise callback) {
+        callback.reject("unsupported", "Retrieving the volume is currently unsupported");
+    }
+
+    @ReactMethod
+    public void setRate(final float rate) {
+        waitForConnection(() -> binder.getPlayback().setRate(rate));
+    }
+
+    @ReactMethod
+    public void getRate(final Promise callback) {
+        waitForConnection(() -> callback.resolve(binder.getPlayback().getRate()));
+    }
+
+    @ReactMethod
+    public void getTrack(final String id, final Promise callback) {
+        waitForConnection(() -> {
+            List<Track> tracks = binder.getPlayback().getQueue();
+
+            for(Track track : tracks) {
+                if(track.id.equals(id)) {
+                    callback.resolve(track); // TODO serialize
+                    return;
+                }
+            }
+
+            callback.reject("unknown", "Track not found");
+        });
+    }
+
+    @ReactMethod
+    public void getCurrentTrack(final Promise callback) {
+        waitForConnection(() -> {
+            Track track = binder.getPlayback().getCurrentTrack();
+
+            if(track == null) {
+                callback.reject("unknown", "Not playing any track");
```

Comment:
ios implementation of getCurrentTrack does not reject if no track is playing since https://github.com/react-native-kit/react-native-track-player/commit/8553021d19a5b5ad5544f416cd5ceae4c5feb5f5 , maybe it should behave in the same way for android? Just resolving with null

Reply:
Good catch, thanks :)